File: 01-Assignment-02-02-2026/Using_C/program_27.c
/* Program 27: Check whether a string is palindrome */

#include <stdio.h>
#include <string.h>
int isPalindrome(char string[]){
    int size = strlen(string);
    int low = 0;
    int high = size -1 ;
    while (low < high){
        if(string[low] != string[high]){
            return 0;
        }
        low++;
        high--;
    }
    return 1;
}

int main() {
    char string[100];
    printf("Enter the String : ");
    scanf("%s",string);
    int res = isPalindrome(string);
    if (res)
        printf("The Given String is Valid Palindrome.");
    else{
        printf("The Given String is Not Valid Palindrome");
    }
    return 0;
}

File: 01-Assignment-02-02-2026/Using_Java/Program27.java
/*
Program 27
Java program to check whether a string is palindrome or not
*/

import java.util.Scanner;

public class Program27 {

    public static boolean isPalindrome(String str) {
        int start = 0;
        int end = str.length() - 1;

        while (start < end) {
            if (str.charAt(start) != str.charAt(end)) {
                return false;
            }
            start++;
            end--;
        }
        return true;
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        System.out.print("Enter a string: ");
        String input = scanner.nextLine();

        // normalize input
        input = input.toLowerCase().replaceAll("\\s+", "");

        if (isPalindrome(input)) {
            System.out.println("It is a Palindrome string!");
        } else {
            System.out.println("It is not a Palindrome string!");
        }

        scanner.close();
    }
}